bofire/utils/torch_tools.py

```diff
@@ -147,24 +148,50 @@ def constrained_objective2botorch(
         objective (BotorchConstrainedObjective): The objective that should be transformed.
 
     Returns:
-        Tuple[List[Callable[[Tensor], Tensor]], List[float]]: List of callables that can be used by botorch for setting up the constrained objective, and
-            list of the corresponding botorch eta values.
+        Tuple[List[Callable[[Tensor], Tensor]], List[float], int]: List of callables that can be used by botorch for setting up the constrained objective,
+            list of the corresponding botorch eta values, final index used by the method (to track for categorical variables)
```

Comment:
For what is this index needed?

Reply:
Since the MLP will output a vector of length `c`, we need to keep track of two items: 1. an index related to the current objective and 2. an index related to the current objective's output. For most other methods, if the output is 1 dimensional, then these two indexes are the same. For the any method that outputs more than one objective value, then eventually these two indexes will be misaligned. 

The output that eventually gets passed to `constrained_objective2botorch` is the output from model.posterior <- which is the result of the forward pass through all models. If one of these models outputs a vector of size > 1, then using solely the objective index results in a failure to capture the underlying model's output. Is this a problem with other methods too? Do I need to add a check for all models somewhere else to measure their output size?

Reply:
Ah, I get it. This makes sense. Currently, I do not see any problem there, but I will think about it ;)